src/main/java/org/gridsuite/securityanalysis/server/repositories/specifications/SpecificationUtils.java

```diff
@@ -137,13 +145,22 @@ private static <X> Specification<X> appendTextFilterToSpecification(Specificatio
 
     @NotNull
     private static <X> Specification<X> appendNumberFilterToSpecification(Specification<X> specification, ResourceFilterDTO resourceFilter) {
-        final double tolerance = 0.00001; // tolerance for comparison
         String value = resourceFilter.value().toString();
-        return createNumberPredicate(specification, resourceFilter, value, tolerance);
+        return createNumberPredicate(specification, resourceFilter, value);
     }
 
-    private static <X> Specification<X> createNumberPredicate(Specification<X> specification, ResourceFilterDTO resourceFilter, String value, double tolerance) {
-        Double valueDouble = Double.valueOf(value);
+    private static <X> Specification<X> createNumberPredicate(Specification<X> specification, ResourceFilterDTO resourceFilter, String filterValue) {
+        // the reference for the comparison is the number of digits after the decimal point in filterValue
+        // extra digits are ignored, but the user may add '0's after the decimal point in order to get a better precision
+        String[] splitValue = filterValue.split("\\.");
+        int numberOfDecimalAfterDot = 0;
+        if (splitValue.length > 1) {
+            numberOfDecimalAfterDot = splitValue[1].length();
+        }
+        // tolerance is multiplied by 0.5 to simulate the fact that the database value is rounded (in the front, from the user viewpoint)
+        // more than 13 decimal after dot will likely cause rounding errors due to double precision
```

Comment:
can we send an error if it happens ?

Reply:
I added a warning on the front side. The user may still try if he wants (it may work after all) but he has been warned.